### PR TITLE
Ensure ClientHandler properly honors Host header

### LINKED ISSUE #1481: System.UriFormatException : Invalid URI: The hostname could not be parsed.
_From @nulltoken on July 2, 2018 12:39_

Hello,

It's possible I may have found a bug (or at the very least a surprising behavior :wink: ).

Running an integration test, using the TestServer, against an Mvc backed API (monitored by ApplicationInsights), throws a `UriFormatException`.

It looks like (cf. the screenshots below) that the host being parsed end up being `"localhost,localhost"`.

I've managed to trim down the code as much as possible into a repro case available at https://github.com/nulltoken/UriFormatException

Beware, that looks pretty random. The repro case breaks most of time with one failing test, and from time to time two tests fail.

FWIW, removing this **[line](https://github.com/nulltoken/UriFormatException/blob/master/test/Tests/Repro.cs#L15-L16)** from the repro case makes the problem disappear.

Versions:
- .Net Core 2.1.1
- ApplicationInsightsAspNetCore 2.4-beta1

hostingapplication

uriext

_Copied from original issue: aspnet/Mvc#7997_

### PR DESCRIPTION
**Pull request by nulltoken**
Fix #1481

From the http client standpoint, a manually set Host header should
always be preferred over the one parsed from the request uri.

### CODE REVIEW

File: src/Microsoft.AspNetCore.TestHost/ClientHandler.cs [hunk 1]
```diff
@@ -13,6 +13,7 @@
 using Microsoft.AspNetCore.Hosting.Server;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Net.Http.Headers;
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.TestHost/ClientHandler.cs]: Why was this needed?

Comment (nulltoken) [reply src/Microsoft.AspNetCore.TestHost/ClientHandler.cs]: @Tratcher Ooops. 

It's a ghost of the shadow of the remnant of a previous code attempt.

The 👻 has just been busted by a team of professionals.

I've been told it shouldn't haunt us back.

**Comment by Tratcher** [comment]
Thanks
